src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -404,74 +358,16 @@ private static string GetCodeFixTitle(string resourceString, IMethodSymbol metho
-            {
-                var document = documentLookup.Key;
-                var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
-                var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-                var editor = new SyntaxEditor(syntaxRoot, solution.Workspace);
-                var generator = editor.Generator;
-                foreach (var methodDeclaration in documentLookup)
-                {
-                    var methodNode = syntaxRoot.FindNode(methodDeclaration.Locations[0].SourceSpan);
-                    var existingParameters = generator.GetParameters(methodNode);
-                    var insertionIndex = isNamedArgument
-                        ? existingParameters.Count
-                        : argumentList.IndexOf(argument);
-
-                    // if the preceding parameter is optional, the new parameter must also be optional 
-                    // see also BC30202 and CS1737
-                    var parameterMustBeOptional = insertionIndex > 0 &&
-                        syntaxFacts.GetDefaultOfParameter(existingParameters[insertionIndex - 1]) != null;
-
-                    var parameterSymbol = CreateParameterSymbol(
-                        methodDeclaration, argumentType, refKind, parameterMustBeOptional, argumentNameSuggestion);
-
-                    var argumentInitializer = parameterMustBeOptional ? generator.DefaultExpression(argumentType) : null;
-                    var parameterDeclaration = generator.ParameterDeclaration(parameterSymbol, argumentInitializer)
-                                                        .WithAdditionalAnnotations(Formatter.Annotation);
-                    if (anySymbolReferencesNotInSource && methodDeclaration == method)
-                    {
-                        parameterDeclaration = parameterDeclaration.WithAdditionalAnnotations(
-                            ConflictAnnotation.Create(FeaturesResources.Related_method_signatures_found_in_metadata_will_not_be_updated));
-                    }
-
-
-                    if (method.MethodKind == MethodKind.ReducedExtension)
-                    {
-                        insertionIndex++;
-                    }
-
-                    AddParameter(
-                        syntaxFacts, editor, methodNode, argument,
-                        insertionIndex, parameterDeclaration, cancellationToken);
-                }
-
-                var newRoot = editor.GetChangedRoot();
-                solution = solution.WithDocumentSyntaxRoot(document.Id, newRoot);
-            }
-
-            return solution;
+            var newParameterIndex = isNamedArgument ? (int?)null : argumentList.IndexOf(argument);
+            return await AddParameterService.Instance
+                .AddParameterAsync(invocationDocument, method, argumentType, refKind, argumentNameSuggestion, newParameterIndex, fixAllReferences, cancellationToken).ConfigureAwait(false);
```

Comment:
can you wrap tehse long lines that don't fit on github? :)  i recommend something more like:

```c#
            return await AddParameterService.Instance.AddParameterAsync(
                invocationDocument, method, argumentType, refKind, argumentNameSuggestion,
                newParameterIndex, fixAllReferences, cancellationToken).ConfigureAwait(false);
```

Reply:
Note: you don't need async/await here as this method no longer does any other awaiting.  You can just directly return the task

Reply:
It does do other awaiting:

`await GetNameSuggestionForArgumentAsync(invocationDocument, argument, cancellationToken).ConfigureAwait(false);`

Reply:
strange, when i looked earlier, it didn't seem like that was there at all.  Can't explain it.  Must have been a brain fart.